alchemy/alchemy.py

```diff
@@ -884,24 +947,32 @@ def _alchemicallyModifyNonbondedForce(self, system, reference_force):
             c_rf = r_cutoff**(-1) * ((3*epsilon_solvent) / (2*epsilon_solvent + 1))
             electrostatics_energy_expression += "k_rf = %f;" % (k_rf.value_in_unit_system(unit.md_unit_system))
             electrostatics_energy_expression += "c_rf = %f;" % (c_rf.value_in_unit_system(unit.md_unit_system))
-        elif method in [openmm.NonbondedForce.PME, openmm.NonbondedForce.Ewald]:
+        elif nonbonded_method in [openmm.NonbondedForce.PME, openmm.NonbondedForce.Ewald]:
             # Ewald direct-space electrostatics
             [alpha_ewald, nx, ny, nz] = reference_force.getPMEParameters()
             if alpha_ewald == 0.0:
                 # If alpha is 0.0, alpha_ewald is computed by OpenMM from from the error tolerance.
                 [alpha_ewald, nx, ny, nz] = reference_force.getPMEParameters()
+                assert (alpha_ewald != 0.0) # we can't have alpha be zero
             electrostatics_energy_expression += "U_electrostatics = (lambda_electrostatics^softcore_d)*ONE_4PI_EPS0*chargeprod*erfc(alpha_ewald*reff_electrostatics)/reff_electrostatics;"
             electrostatics_energy_expression += "alpha_ewald = %f;" % (alpha_ewald.value_in_unit_system(unit.md_unit_system))
             # TODO: Handle reciprocal-space electrostatics for alchemically-modified particles.  These are otherwise neglected.
             # NOTE: There is currently no way to do this in OpenMM.
         else:
-            raise Exception("Nonbonded method %s not supported yet." % str(method))
+            raise Exception("Nonbonded method %s not supported yet." % str(nonbonded_method))
 
         # Add additional definitions common to all methods.
         sterics_energy_expression += "reff_sterics = sigma*((softcore_alpha*(1.-lambda_sterics)^softcore_b + (r/sigma)^softcore_c))^(1/softcore_c);" # effective softcore distance for sterics
```

Comment:
This is really just a formatting issue, but can we declare floats as `1.0` instead of just `1.`? It can get a bit confusing from a human readable perspective.